plugins/modules/inventory_intent.py

```diff
@@ -961,10 +1089,162 @@ def get_diff_merged(self, config):
+                        self.status = "success"
+                        self.result['changed'] = True
+                        self.result['response'] = execution_details
+                        break
+                    elif execution_details.get("isError") and execution_details.get("failureReason"):
+                        self.msg = "AP Device Reboot get failed because of {0}".format(execution_details.get("failureReason"))
+                        self.status = "failed"
+                        break
+                self.log("AP Devices Rebooted Successfully and Rebooted devices are :" + str(device_ips))
+                msg = "Device " + str(device_ips) + " Rebooted Successfully !!"
+                self.result['log'].append(msg)
+
+            return self
+
+        if self.config[0].get('export_device_list'):
+
+            device_ips = self.config[0].get("ip_address")
+            device_uuids = []
+            try:
+                for device_ip in device_ips:
+                    response = self.dnac._exec(
+                        family="devices",
+                        function='get_device_list',
+                        params={"managementIpAddress": device_ip}
+                    )
+                    response = response.get('response')
+                    if len(response) > 0:
+                        device_uuids.append(response[0]["id"])
+
+                # Now all device UUID get collected so call the export device list API
+
+                export_device_details = self.config[0].get('export_device_list')
+                payload_params = {
+                    "deviceUuids": device_uuids,
+                    "password": export_device_details.get("password"),
+                    "operationEnum": export_device_details.get("operation_enum", "1"),
+                    "paramters": export_device_details.get("paramters")
+                }
+                response = self.dnac._exec(
+                    family="devices",
+                    function='export_device_list',
+                    op_modifies=True,
+                    params=payload_params,
+                )
+                self.log(str(response))
+                response = response.get("response")
+                task_id = response.get("taskId")
+                # With this task ID call the Get Task Details API
+                task_resp = self.dnac._exec(
+                    family="task",
+                    function='get_task_by_id',
+                    op_modifies=True,
+                    params={"task_id": task_id},
+                )
+
+                while True:
+                    execution_details = self.get_task_details(task_id)
+
+                    if execution_details.get("additionalStatusURL"):
+                        file_id = execution_details.get("additionalStatusURL").split("/")[-1]
+                        break
+                    elif execution_details.get("isError") and execution_details.get("failureReason"):
```

Comment:
and check of execution_details.get("failureReason"): not required. Also we collect failed_reason in next line. you can follow the above comments.

Reply:
updated at all places.